allensdk/experimental/nwb/ic_ephys.py

```diff
@@ -0,0 +1,176 @@
+        sweep_metadata["stimulus_scale_factor"] = scale_factor
+        cnt = notebook.get_value("Set Sweep Count", sweep_number, 0)
+        stim_code_ext = acquisition["stimulus_description"] + "[%d]" % int(cnt)
+        sweep_metadata["stimulus_code_ext"] = stim_code_ext
+
+        # ------------------------------------------------------------------------------
+
+        if stimulus["clamp_mode"] == "voltage_clamp":
+
+            stimulus_series = VoltageClampStimulusSeries(
+                name=sweep_name, source=stimulus['source'], data=stimulus['data'], unit=stimulus['unit'],
+                conversion = stimulus['conversion'],
+                starting_time=np.nan,
+                electrode=elec, gain=np.nan,
+                rate=stimulus['rate'],
+            )
+        elif stimulus["clamp_mode"] == "current_clamp":
+
+            stimulus_series = CurrentClampStimulusSeries(
+                name=sweep_name, source=stimulus['source'], data=stimulus['data'], unit=stimulus['unit'],
+                conversion = stimulus['conversion'],
+                starting_time=np.nan,
+                electrode=elec, gain=np.nan,
+                rate=stimulus['rate'],
+            )
+
+        if acquisition["clamp_mode"] == "voltage_clamp":
+
+            acquisition_series = VoltageClampSeries(
+                name=sweep_name, source=acquisition['source'], data=acquisition['data'],
+                unit=acquisition['unit'], conversion=acquisition['conversion'],
+                resolution=np.nan, starting_time=np.nan, rate=acquisition['rate'],
+                electrode=elec, gain=np.nan, capacitance_slow=np.nan, resistance_comp_correction=np.nan,
+                capacitance_fast=np.nan, resistance_comp_bandwidth=np.nan, resistance_comp_prediction=np.nan,
+                whole_cell_capacitance_comp=np.nan, whole_cell_series_resistance_comp=np.nan,
+                stimulus_description=acquisition["stimulus_description"]
+            )
+        elif acquisition["clamp_mode"] == "current_clamp":
+
+            bridge_balance = notebook.get_value("Bridge Bal Value", sweep_number, np.nan)
+            bias_current = notebook.get_value("I-Clamp Holding Level", sweep_number, np.nan)
+
+            acquisition_series = CurrentClampSeries(
+                name=sweep_name, source=acquisition['source'], data=acquisition['data'],
+                unit=acquisition['unit'], conversion=acquisition['conversion'], resolution=np.nan, starting_time=np.nan,
+                rate=acquisition['rate'],
+                electrode=elec, gain=np.nan,
+                bias_current=bias_current,
+                bridge_balance=bridge_balance,
+                stimulus_description=acquisition["stimulus_description"],
+                capacitance_compensation=np.nan
+            )
+
+        nwb2file.add_stimulus(stimulus_series)
+        nwb2file.add_acquisition(acquisition_series)
+
+    return nwb2file
+
+
+def build_sweep_table(nwbfile):
```

Comment:
This looks great.  Would it make sense to add references to these sweeps using an epoch table?  Now that @NileGraddis has his work merged and it round-trips, and it might enhance this work.  Here is the code that I wrote to generate a table from my data, that you could use for starting: https://github.com/AllenInstitute/AllenSDK/blob/6b978666b8a7ced41e783b7b95ce072e1b55105e/allensdk/experimental/nwb/stimulus.py#L124

Reply:
Thanks for bringing this up. I should definitely look into this in the near future.